### PR TITLE
Add pull request template with checklist

### PR DESCRIPTION
**Pull request by PVince81**
as discussed with @DaphneMuller and others

borrowed from the deck app

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+* Resolves: # <!-- related github issue -->
+* Target version: master 
+
+### Summary
```
Comment (ChristophWurst) [inline .github/pull_request_template.md:5]:
```suggestion
## Summary
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+* Resolves: # <!-- related github issue -->
+* Target version: master
```
Comment (ChristophWurst) [inline .github/pull_request_template.md:3]: That is always the case, isn't it?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+* Resolves: # <!-- related github issue -->
+* Target version: master 
+
+### Summary
+
+
+### TODO
```
Comment (ChristophWurst) [inline .github/pull_request_template.md:8]:
```suggestion
## TODO
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+- [ ] ...
+
+### Checklist
```
Comment (ChristophWurst) [inline .github/pull_request_template.md:12]:
```suggestion
## Checklist
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+- [ ] Code is properly formatted
+- [ ] Sign-off message is added to all commits
+- [ ] Tests (unit, integration, api and/or acceptance) are included
+- [ ] Documentation (manuals or wiki) has been updated or is not required
```
Comment (ChristophWurst) [inline .github/pull_request_template.md:17]: to help new contributors we could provide links to documentation of these aspects, e.g.

1) Link to the documentation that describes cs-fixer and eslint in server and how it's run
2) Link to the DCO documentation
3) Link to the documentation that describes how server unit and integration tests are run
4) Link to https://github.com/nextcloud/documentation

**Review by ChristophWurst — COMMENTED** [review]
Nitpicks: heading hierarchy & making it easier for new contributors

File: .github/pull_request_template.md (hunk 2 above)
Comment (juliushaertl) [reply .github/pull_request_template.md:3]: Yep, I also removed this when drafting the templates for our team apps

Comment (PVince81) [reply .github/pull_request_template.md:3]: okay, I forgot to pull when taking the template from deck.
I'll pull again

Comment (PVince81) [reply .github/pull_request_template.md:3]: hmmm, I just pulled on the deck repo and it's the same https://github.com/nextcloud/deck/blob/master/.github/pull_request_template.md

maybe we add a template in https://github.com/nextcloud/.github and use it everywhere then ?

File: .github/pull_request_template.md (hunk 5 above)
Comment (PVince81) [reply .github/pull_request_template.md:17]: in theory we wouldn't need checkboxes for everything that is covered by CI

still, good to have a checkbox for adding new tests

**Comment by PVince81** [comment]
updated, please re-review

**Review by ChristophWurst — APPROVED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (juliushaertl) [reply .github/pull_request_template.md]: I just drafted it locally, pushed my version with taking the screenshot template from spreed to https://github.com/nextcloud/text/pull/3424

But I'd agree that we should probably just have a template in the .github repo

**Comment by juliushaertl** [comment]
One topic @max-nextcloud brought up when discussing was that checkboxes for anything that we have as CI checks might not make much sense as they tend to not represent the latest state. Maybe we should just have those as a list of things to check on the github actions with links for hints?

**Comment by PVince81** [comment]
@juliushaertl adjusted, please review

**Review by juliusknorr — APPROVED** [review]

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+## TODO
+
+- [ ] ...
+
+## Checklist
+
+- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
+- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
+- Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
```
Comment (nickvergessen) [inline .github/pull_request_template.md:15]: But this makes sense as a checkbox? It's not about whether it passed, but if new ones where added

Comment (PVince81) [reply .github/pull_request_template.md:15]: argh, I misthought

Comment (PVince81) [reply .github/pull_request_template.md]: adjusted

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**